completion.go

```diff
@@ -10,15 +10,23 @@ import (
 
 const completionFlag = "--mcli-generate-completion"
 
-func hasCompletionFlag(args []string) (bool, []string) {
-	var lastArg string
-	if len(args) > 0 {
-		lastArg = args[len(args)-1]
-	}
-	if lastArg == completionFlag {
-		return true, args[:len(args)-1]
+func getAllowedShells() []string {
+	return []string{"bash", "zsh", "fish", "powershell"}
+}
+
+func hasCompletionFlag(args []string) (bool, []string, string) {
+	shell := "unsupported"
+	foundIndex := find(args, completionFlag)
+	if len(args) >= 2 && foundIndex >= 0 {
+		proposedShell := args[foundIndex+1]
```

Comment:
In case the last argument is "--mcli-generate-completion", `foundIndex` would be `len(args)-1`, here `args(foundIndex+1]` -> `args[len(args)]` will panic.

Reply:
Yes, I agree. Addressed in fix and unit test.